libsolidity/formal/SMTChecker.cpp

```diff
@@ -47,6 +47,19 @@ SMTChecker::SMTChecker(ErrorReporter& _errorReporter, map<h256, string> const& _
 #endif
 }
 
+SMTChecker::~SMTChecker()
+{
+#if !(defined (HAVE_Z3) || defined (HAVE_CVC4))
```

Comment:
Perhaps add another one for the smtlib case?

Reply:
Not sure I get it, the SmtLib2Interface is always instantiated.

Reply:
I'd display a similar, but slightly different, message for smtlib.